vendors/nordic/boards/nrf52840-dk/ports/common_io/iot_flash.c

```diff
@@ -0,0 +1,490 @@
+#define FLASH_PADDING_BYTE      0xFF       /* NAND-flash default logic value */
+
+static void prvFlashEventHandler( nrf_fstorage_evt_t * p_evt );
+
+NRF_FSTORAGE_DEF( nrf_fstorage_t iot_fstorage ) =
+{
+    /* Set a handler for fstorage events. */
+    .evt_handler = prvFlashEventHandler,
+};
+
+typedef enum
+{
+    IOT_FLASH_CLOSED = 0u,
+    IOT_FLASH_OPENED = 1u
+} IotFlashState_t;
+
+/* @brief flash context */
+typedef struct IotFlashDescriptor
+{
+    IotFlashInfo_t xFlashInfo;    /* flash info structure */
+    IotFlashCallback_t xCallback; /* callback */
+    void * pvUserContext;         /* user context to provide in callback */
+    uint8_t ucState;
+} IotFlashDescriptor_t;
+
+#define DEFAULT_FLASH_DESC                          \
+    ( IotFlashDescriptor_t )                        \
+    {                                               \
+        .xFlashInfo =                               \
+        {                                           \
+            .ulFlashSize       = 0,                 \
+            .ulBlockSize       = FLASH_BLOCK_SIZE,  \
+            .ulSectorSize      = FLASH_SECTOR_SIZE, \
+            .ulPageSize        = FLASH_PAGE_SIZE,   \
+            .ulLockSupportSize = 0,                 \
+            .ucAsyncSupported  = 0,                 \
+        },                                          \
+        .xCallback = NULL,                          \
+        .pvUserContext = NULL,                      \
+        .ucState = IOT_FLASH_CLOSED                 \
+    }
+
+static IotFlashDescriptor_t xFlashDesc = DEFAULT_FLASH_DESC;
+
+/*---------------------------------------------------------------------------------*
+*                                Private Helpers                                  *
+*---------------------------------------------------------------------------------*/
+static inline bool prvIsAligned4K( uint32_t ulAddr )
+{
+    return ( ulAddr & FLASH_SECTOR_MASK_4K ) == 0;
+}
+
+static inline void prvWaitUntilFlashReady( nrf_fstorage_t const * p_fstorage )
+{
+    while( nrf_fstorage_is_busy( p_fstorage ) )
+    {
+        vTaskDelay( pdMS_TO_TICKS( 1 ) );
+    }
+}
+
+static void prvFlashEventHandler( nrf_fstorage_evt_t * p_evt )
+{
+    IotFlashOperationStatus_t xStatus = IOT_FLASH_SUCCESS;
+
+    if( p_evt->result != NRF_SUCCESS )
+    {
+        switch( p_evt->id )
+        {
+            case NRF_FSTORAGE_EVT_WRITE_RESULT:
+                xStatus = IOT_FLASH_WRITE_FAILED;
+                break;
+
+            case NRF_FSTORAGE_EVT_ERASE_RESULT:
+                xStatus = IOT_FLASH_ERASE_FAILED;
+                break;
+
+            default:
+                break;
+        }
+    }
+
+    if( xFlashDesc.xCallback != NULL )
+    {
+        xFlashDesc.xCallback( xStatus, xFlashDesc.pvUserContext );
+    }
+}
+
+static void prvGetFlashInfo( void )
+{
+    uint32_t const bootloader_addr = NRF_UICR->NRFFW[ 0 ];
+    uint32_t const page_sz = NRF_FICR->CODEPAGESIZE;
```

Comment:
These variables go against the coding style, I think common-io uses the FreeRTOS coding style, instead of the C-SDK

Reply:
Thanks. Addressed in latest commit